lib/services/address/index.js

```diff
@@ -397,6 +407,13 @@ AddressService.prototype.updateMempoolIndex = function(tx, add, callback) {
       value: inputValue
     });
 
+    var addressIndexKey = encoding.encodeMempoolAddressIndexKey(inputHashBuffer, inputHashType);
```

Comment:
Good catch.

What about a counter? `add` will  +1 the index, and remove will -1 the
index.

Or, in order be less fragile, we can do: index[address] = [txids,...] and
keep the list of related txids.

If add will push txids, remove will delete them. And we check for array
size.

what you think?

On Wed, Feb 10, 2016 at 10:22 AM, Braydon Fuller <notifications@github.com>
wrote:

> In lib/services/address/index.js
> <https://github.com/bitpay/bitcore-node/pull/407#discussion_r52471014>:
>
> > @@ -397,6 +407,13 @@ AddressService.prototype.updateMempoolIndex = function(tx, add, callback) {
> >        value: inputValue
> >      });
> >
> > +    var addressIndexKey = encoding.encodeMempoolAddressIndexKey(inputHashBuffer, inputHashType);
>
> We may need to have these be separate indexes. If there are two
> transactions in the mempool for an address where one is receiving to an
> address and the other spending from an address, and one of these
> transactions is removed from the mempool (was included in a block). The
> index will be removed and then query for the still in mempool transaction
> would be skipped.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/bitpay/bitcore-node/pull/407/files#r52471014>.
>



-- 
Matías Alejo Garcia
@ematiu
Roads? Where we're going, we don't need roads!